include/eve/memory/aligned_ptr.hpp

```diff
@@ -308,4 +309,17 @@ namespace eve
   struct  pointer_alignment<aligned_ptr<Type, Alignment>>
         : std::integral_constant<std::size_t,Alignment>
   {};
+
+  template <typename T, typename Cardinal>
+  EVE_FORCEINLINE auto previous_aligned_address(T* p, Cardinal /*width*/) noexcept
+  {
+    static constexpr std::size_t alignment = Cardinal() * sizeof(T);
```

Comment:
not sure the static is required here.
`constinit` maybe if we really want to hammer down its pure compile-time nature ?
Does clang supports it ?

Reply:
Oh - sorry missed it. `static constexpr` doesn't do anything. On the other hand plain `constexpr` tends to leave garbage on the stack for whatever reason.

Can't find an example but you can see it does not harm:
https://godbolt.org/z/46ojM9